test/TesterInternal/SchedulerTests/OrleansTaskSchedulerAdvancedTests_Set2.cs

```diff
@@ -604,19 +604,32 @@ public void ActivationSched_SubTaskExecutionSequencing()
             {
                 TimeSpan waitCheckTime = TimeSpan.FromMilliseconds(150 * ChainLength * NumChains * waitFactor);
 
-                var timeoutLimit = TimeSpan.FromSeconds(1);
                 try
                 {
                     await resultHandles[i].Task.WithTimeout(waitCheckTime);
                 }
                 catch (TimeoutException)
                 {
-                    Assert.True(false, "Result did not arrive before timeout " + timeoutLimit);
+                    Assert.True(false, "Result did not arrive before timeout " + waitCheckTime);
                 }
 
                 bool ok = resultHandles[i].Task.Result;
-                Assert.False(taskChainEnds[i].IsFaulted, "Task chain " + i + " should not be Faulted: " + taskChainEnds[i].Exception);
+                
+                if (!taskChainEnds[i].IsCompleted)
```

Comment:
since the WithTimeout extension method is more involved than I'd like, I wanted to keep this loop lightweight. But in hindsight it probably doesn't matter, I'll go with your suggestion